pkg/store/memory_store.go

```diff
@@ -161,3 +179,83 @@ func (s *InMemoryStore) GetUpdates() []upstreamtypes.ChannelRelease {
 func (s *InMemoryStore) SetUpdates(updates []upstreamtypes.ChannelRelease) {
 	s.updates = updates
 }
+
+func (s *InMemoryStore) CreateInstanceReportEvent(event heartbeattypes.InstanceReportEvent) error {
```

Comment:
i don't think this belongs in the "memory" store. my suggestion would be to move this logic to the `heartbeat` package. maybe in a `instance_report.go` file.